src/pretix/plugins/stripe/payment.py

```diff
@@ -110,41 +110,40 @@ def payment_perform(self, request, order) -> str:
             else:
                 err = {'message': str(e)}
                 logger.exception('Stripe error: %s' % str(e))
-            messages.error(request, _('Stripe reported an error with your card: %s' % err['message']))
             logger.info('Stripe card error: %s' % str(err))
             order.payment_info = json.dumps({
                 'error': True,
                 'message': err['message'],
             })
             order.save()
+            raise PaymentException(_('Stripe reported an error with your card: %s' % err['message']))
```

Comment:
This needs to be ``raise PaymentException(_('Stripe reported an error with your card: %s') % err['message'])`` instead. The reason is, that we *first* need to pass the string ``'Stripe reported an error with your card: %s'`` to the translation system to have it translated in a different language and *then* insert the custom data.